Sources/Apodini/Metadata/BuiltinMetadata/CommunicationalPatternMetadata.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  CommunicationalPatternMetadata.swift
+//  
+//
+//  Created by Max Obermeier on 30.06.21.
+//
+
+import Foundation
+
+/// A type describing the kind of communication that a ``Handler`` is designed for.
+///
+/// The pattern in a communication is defined by the quantity and order of messages
+/// that are sent from client to server and from server to client.
+public enum CommunicationalPattern {
+    /// **One** client message followed by **one** service message
+    case requestResponse
+    /// **Any amount** of client messages followed by **one** service message
+    case clientSideStream
+    /// **One** client message followed by **any amount** of service messages
+    case serviceSideStream
+    /// **Any amount** of client messages and **any amount** of service messages in an **undefined order**
+    case bidirectionalStream
+}
+
+
+public struct CommunicationalPatternContextKey: OptionalContextKey {
+    public typealias Value = CommunicationalPattern
+}
+
+extension HandlerMetadataNamespace {
+    /// Name Definition for the `ComponentDescriptionMetadata`
+    public typealias Pattern = CommunicationalPatternMetadata
+}
+
+/// The ``CommunicationalPatternMetadata`` can be used to explicitly declare a ``Handler``'s
+/// ``CommunicationalPattern``.
+///
+/// The framework can automatically detect ``CommunicationalPattern/requestResponse`` if the
+/// ``Handler/handle()-94skg``'s return type is no ``Response``. If that is not the case and you don't
+/// explicitly specify the ``HandlerMetadataNamespace/Pattern``, the ``Handler`` is assumed to use
+/// ``CommunicationalPattern/bidirectionalStream``.
```

Comment:
I don't quite get what you mean. Both reads the same for me. If the response does not make use of `Response`, we assume it to be `.requestResponse`, otherwise we assume the "worst" which is `.bidirectionalStream`. Note that there is a **no** in "return type is **no** ``Response``".

Reply:
just noticed I overhead the whole first sentence 😅 Seemingly got a bit too confused by the 94skg 😂